python/cudf/cudf/_fuzz_testing/csv.py

```diff
@@ -100,9 +158,39 @@ def generate_input(self):
             f"Generating DataFrame with rows: {num_rows} "
             f"and columns: {num_cols}"
         )
-        df = cudf.DataFrame.from_arrow(
-            dg.rand_dataframe(dtypes_meta, num_rows, seed)
-        )
-        logging.info(f"Shape of DataFrame generated: {df.shape}")
+        table = dg.rand_dataframe(dtypes_meta, num_rows, seed)
+        df = pyarrow_to_pandas(table)
 
+        logging.info(f"Shape of DataFrame generated: {df.shape}")
+        self._current_buffer = df
         return df
+
+    def write_data(self, file_name):
+        if self._current_buffer is not None:
+            self._current_buffer.to_csv(file_name + "_crash.csv")
+
+    def get_rand_params(self, params):
+        params_dict = {}
+        for param, values in params.items():
+            if param == "columns" and values is None:
+                col_size = self._rand(len(self._current_buffer.columns))
+                params_dict[param] = list(
+                    np.unique(
+                        np.random.choice(
+                            self._current_buffer.columns, col_size
+                        )
+                    )
+                )
+            elif param == "chunksize" and values is None:
+                params_dict[param] = np.random.choice(
+                    [
+                        None,
+                        np.random.randint(
+                            low=1, high=max(1, len(self._current_buffer))
```

Comment:
If the buffer is empty this looks like it would cause problems

Reply:
This would still work with empty dataframe no?
```python
>>> import cudf
>>> df = cudf.DataFrame((
... ))
>>> df
Empty DataFrame
Columns: []
Index: []
>>> df.to_csv(chunksize=1)
'\n'
```

Reply:
Sorry, brainfart on my end, read `max` as `min` 😄